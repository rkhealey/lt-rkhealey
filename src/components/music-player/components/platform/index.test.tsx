import React from 'react';
import { noop } from 'lodash';
import { shallow } from 'enzyme';
import { mocked } from 'ts-jest/utils';
import Platform from '.';
import fetchImage from './fetch-image';

const mockData = [
  {
    id: 'spotify',
    name: 'Spotify',
    url: 'https://open.spotify.com/track/7pKfPomDEeI4TPT6EOYjn9?si=JVKZYLN8RdqzxIFZ_NwhfQ',
    songId: '7pKfPomDEeI4TPT6EOYjn9',
  },
  {
    id: 'extra-platform',
    name: 'Extra Platform',
    url: 'https://extraplatform.com',
    songId: '7pKfPomDEeI4TPT6EOYjn9',
  },
];

jest.spyOn(React, 'useEffect').mockImplementationOnce((f) => f());
jest.mock('./fetch-image');
const mockedFetch = mocked(fetchImage, true);
mockedFetch.mockResolvedValue('/path-to-image.svg');

describe('Platform', () => {
  it('Renders', async () => {
    const wrapper = shallow(<Platform {...mockData[0]} onLaunchPlayer={noop} />);
    // wait for useEffect to finish
    await Promise.resolve();
    expect(wrapper).toMatchSnapshot();
  });

  it('Returns null if logo is not found', () => {
    const wrapper = shallow(<Platform {...mockData[1]} onLaunchPlayer={noop} />);
    expect(wrapper).toEqual({});
  });
});
