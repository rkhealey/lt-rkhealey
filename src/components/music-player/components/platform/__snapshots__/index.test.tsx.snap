// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Platform Renders 1`] = `
<Styled(styled.div)
  alignItems="center"
  justifyContent="space-between"
  key="https://open.spotify.com/track/7pKfPomDEeI4TPT6EOYjn9?si=JVKZYLN8RdqzxIFZ_NwhfQ"
>
  <Styled(styled.div)
    alignItems="center"
    px={3}
  >
    <ForwardRef
      as="a"
      href="https://open.spotify.com/track/7pKfPomDEeI4TPT6EOYjn9?si=JVKZYLN8RdqzxIFZ_NwhfQ"
      target="_blank"
      variant="transparent"
    >
      <ForwardRef
        src="/path-to-image.svg"
      />
    </ForwardRef>
  </Styled(styled.div)>
  <Styled(styled.div)
    alignItems="center"
    flex={1}
    justifyContent="space-between"
    py={3}
    variant="withBorderBottom"
  >
    <ForwardRef>
      Spotify
    </ForwardRef>
    <ForwardRef
      mr={3}
      onClick={[Function]}
      variant="transparent"
    >
      <i
        className="material-icons"
      >
        chevron_right
      </i>
    </ForwardRef>
  </Styled(styled.div)>
</Styled(styled.div)>
`;
